cmd/gcp-controller-manager/node_csr_approver_test.go

```diff
@@ -1640,3 +1644,194 @@ func TestParseInstanceGroupURL(t *testing.T) {
+					Conditions: []v1.PodCondition{
+						{
+							Type:   v1.PodReady,
+							Status: v1.ConditionTrue,
+						},
+					},
+				},
+			},
+			expectedPatchedPod:   nil,
+			expectedDeleteAction: &testclient.DeleteActionImpl{Name: "testPod", DeleteOptions: metav1.DeleteOptions{GracePeriodSeconds: &zero}},
+		},
+		{
+			desc: "Pod that was in Succeeded phase should remain in Succeeded phase prior to deletion. Pod should not be patched because it is already in terminal phase.",
+			node: testNode,
+			pod: &v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "default",
+					Name:      "testPod",
+				},
+				Spec: v1.PodSpec{
+					NodeName: "testNode",
+				},
+				Status: v1.PodStatus{
+					Phase: v1.PodSucceeded,
+					Conditions: []v1.PodCondition{
+						{
+							Type:   v1.PodReady,
+							Status: v1.ConditionTrue,
+						},
+					},
+				},
+			},
+			expectedPatchedPod:   nil,
+			expectedDeleteAction: &testclient.DeleteActionImpl{Name: "testPod", DeleteOptions: metav1.DeleteOptions{GracePeriodSeconds: &zero}},
+		},
+	} {
+		t.Run(tc.desc, func(t *testing.T) {
+
+			nodeList := &v1.NodeList{}
+			nodeList.Items = append(nodeList.Items, *tc.node)
+			podList := &v1.PodList{}
+			podList.Items = append(podList.Items, *tc.pod)
+
+			client := fake.NewSimpleClientset(nodeList, podList)
+			controllerCtx := &controllerContext{client: client}
+
+			err := deleteAllPodsBoundToNode(controllerCtx, tc.node.Name)
+
+			if err != nil {
+				t.Fatalf("Unexpected error deleting all pods bound to node %v", err)
+			}
+
+			actions := client.Actions()
+
+			expectedNumberOfActions := 0
+			if tc.expectedPatchedPod != nil {
+				expectedNumberOfActions++
+			}
+			if tc.expectedDeleteAction != nil {
+				expectedNumberOfActions++
+			}
+			// Always expect a list pods action
+			expectedNumberOfActions++
+
+			if len(actions) != expectedNumberOfActions {
+				t.Fatalf("Unexpected number of actions, got %v, want 3 (list pods, patch pod, delete pod)", len(actions))
+			}
+
+			var patchAction testclient.PatchAction
+			var deleteAction testclient.DeleteAction
+
+			for _, action := range actions {
+				if action.GetVerb() == "patch" {
+					patchAction = action.(testclient.PatchAction)
+				}
+
+				if action.GetVerb() == "delete" {
+					deleteAction = action.(testclient.DeleteAction)
+				}
+			}
+
+			if tc.expectedPatchedPod != nil {
+				patchedPodBytes := patchAction.GetPatch()
+
+				originalPod, err := json.Marshal(tc.pod)
+				if err != nil {
+					t.Fatalf("Failed to marshal original pod %#v: %v", originalPod, err)
+				}
+				updated, err := strategicpatch.StrategicMergePatch(originalPod, patchedPodBytes, v1.Pod{})
+				if err != nil {
+					t.Fatalf("Failed to apply strategic merge patch %q on pod %#v: %v", patchedPodBytes, originalPod, err)
+				}
+
+				updatedPod := &v1.Pod{}
+				if err := json.Unmarshal(updated, updatedPod); err != nil {
+					t.Fatalf("Failed to unmarshal updated pod %q: %v", updated, err)
+				}
```

Comment:
I wish this was part of the k8s testing helpers :sweat_smile:

@aojea WDYT?

Reply:
It would be nice to be included in testing helpers :)